tiledb/tests/test_vfs.py

```diff
@@ -217,6 +217,28 @@ def test_io(self):
         self.assertEqual(fio.readinto(test_bytes), 10)
         self.assertEqual(test_bytes, buffer)
 
+        # Test readinto with random np.float32
+        fio.seek(0)
+        test_np_array = np.empty(10, dtype=np.float32)
+        n_bytes = fio.readinto(test_np_array)
+        self.assertEqual(n_bytes, 10)
+        self.assertEqual(test_np_array.tobytes()[:n_bytes], buffer)
+
+        # Test readinto with np.int32 buffer trying to write bytes that cannot be converted to int32
+        buffer = b"012\x00\x01"
+        with tiledb.FileIO(vfs, self.path("foo"), mode="wb") as fio:
+            fio.write(buffer)
+            fio.flush()
+            self.assertEqual(fio.tell(), len(buffer))
+
+        fio = tiledb.FileIO(vfs, self.path("foo"), mode="rb")
+
+        fio.seek(0)
+        test_np_array = np.empty(5, dtype=np.int32)
+        n_bytes = fio.readinto(test_np_array)
+        self.assertEqual(n_bytes, 5)
+        self.assertEqual(test_np_array.tobytes()[:n_bytes], buffer)
```

Comment:
@kounelisagis what would you think about pulling this out into a separate test function? Right now `test_io()` is a bunch of different tests that are always run in the same order and are stateful. This can hide subtle bugs.

Reply:
done.